src/main/java/io/cryostat/rules/PeriodicArchiver.java

```diff
@@ -146,4 +181,17 @@ public void run() {
         }
         return future;
     }
+
+    public Future<JsonArray> getArchivedRecordings() throws Exception {
+
+        CompletableFuture<JsonArray> future = new CompletableFuture<>();
```

Comment:
I'm not sure why this needs to return a `Future` - can't `run()` above just directly call `recordingArchiveHelper.getRecordings()`? Then above the direct return type (currently `List<Map<String, String>>`, but as I said in an earlier comment, should be `List<SomeOtherType>`) can be used, rather than having to go through this JSON encode/decode cycle on top.

Reply:
For the `Future`, I was following the advice here: 

https://github.com/cryostatio/cryostat/pull/557#discussion_r665415792

Should I change it to return a non-Future object? 

The separate method does seem redundant though, especially with the JSON overhead so I'll change that.

Reply:
Oh, my bad - I think when I wrote that I was thinking that the method belonged to one of the `Helper`s. Since those are part of an internal API, it makes sense for them to be public and to return a `Future` to indicate to callers that these are blocking operations. This `getArchivedRecordings` method and others within the `PeriodicArchiver` are not really part of an internal API, so it's okay for them to be non-`public`, and in that case certainly don't need to return `Future`s or otherwise be marked as long-running operations if they are just internal methods within the class.

Reply:
So the actual advice I meant to share in that comment would mean that in this case, `RecordingArchiveHelper#getRecordings` should be `public`, and should have a return type of `Future<List<ArchivedRecordingInfo>>` (or whatever that type to replace the `Map<String, String>` gets named).

Reply:
No worries, I'll go ahead and remove the unnecessary `Future` stuff from the `PeriodicArchiver`.